akka-actor/src/main/scala/akka/dispatch/affinity/AffinityPool.scala

```diff
@@ -28,6 +28,84 @@ import akka.util.OptionVal
 import scala.collection.mutable
 import scala.util.control.NonFatal
 
+@InternalApi
+@ApiMayChange
+private[affinity] object AffinityPool {
+  type PoolState = Int
+  // PoolState: accepts new tasks and processes tasks that are enqueued
+  final val Running = 0
+  // PoolState: does not accept new tasks, processes tasks that are in the queue
+  final val ShuttingDown = 1
+  // PoolState: does not accept new tasks, does not process tasks in queue
+  final val ShutDown = 2
+  // PoolState: all threads have been stopped, does not process tasks and does not accept new ones
+  final val Terminated = 3
+
+  // Method handle to JDK9+ onSpinWait method
+  private val onSpinWaitMethodHandle =
+    try
+      OptionVal.Some(MethodHandles.lookup.findStatic(classOf[Thread], "onSpinWait", methodType(classOf[Unit])))
+    catch {
+      case NonFatal(_) ⇒ OptionVal.None
+    }
+
+  type IdleState = Int
+  // IdleState: Initial state
+  final val Initial = 0
+  // IdleState: Spinning
+  final val Spinning = 1
+  // IdleState: Yielding
+  final val Yielding = 2
+  // IdleState: Parking
+  final val Parking = 3
+
+  // Following are auxiliary class and trait definitions
+  private final class IdleStrategy(val idleCpuLevel: Int) {
+
+    private[this] val maxSpins = 1100 * idleCpuLevel - 1000
```

Comment:
@Hawstein tgis has been borrowed from https://github.com/akka/akka/blame/master/akka-remote/src/main/scala/akka/remote/artery/TaskRunner.scala#L99 where there are some additional comments, but I assume that this is based on trying out a few different magic numbers.  @patriknw should be able to share more as he is the one who wrote it. Nevertheless maybe we can provide more parameters to allow users to finetune their backoff strategy in order to achieve more bespoke behaviour and  fit their message rate rhythm.